data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaSourceCustomConsumer.java

```diff
@@ -122,39 +132,45 @@ public void updateOffsetsToCommit(final TopicPartition partition, final OffsetAn
         }
     }
 
-    private AcknowledgementSet createAcknowledgementSet(Map<TopicPartition, Range<Long>> offsets) {
+    private AcknowledgementSet createAcknowledgementSet(Map<TopicPartition, CommitOffsetRange> offsets) {
         AcknowledgementSet acknowledgementSet =
-            acknowledgementSetManager.create((result) -> {
-                if (result == true) {
-                    topicMetrics.getNumberOfPositiveAcknowledgements().increment();
-                    synchronized(acknowledgedOffsets) {
-                        acknowledgedOffsets.add(offsets);
+                acknowledgementSetManager.create((result) -> {
+                    numberOfAcksPending.decrementAndGet();
+                    if (result == true) {
+                        topicMetrics.getNumberOfPositiveAcknowledgements().increment();
+                        synchronized(this) {
```

Comment:
Minor: I think we could use the built-in concurrent collection implementations rather than manually synchronizing